src/highdicom/seg/sop.py

```diff
@@ -2150,7 +2150,17 @@ def _get_ref_instance_uids(self) -> List[Tuple[str, str, str]]:
                             )
                         )
 
-        return instance_data
+        # There shouldn't be duplicates here, but there's no explicit rule
+        # preventing it.
+        # Since dictionary ordering is preserved, this trick deduplicates
+        # the list without changing the order
+        unique_instance_data = list(dict.fromkeys(instance_data))
+        if len(unique_instance_data) != len(instance_data):
+            logger.warning(
+                'Duplicate entries found in the ReferencedSeriesSequence.'
```

Comment:
This log message could benefit from a few additional details, such as the SOP Instance UID of the instance and potentially also the duplicate entries. That information may help developers to debug the issue.

Reply:
Addressed in 311d2f0